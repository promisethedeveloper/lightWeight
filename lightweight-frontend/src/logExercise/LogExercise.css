.LogExercise-edit-button {
	color: #461111;
}
