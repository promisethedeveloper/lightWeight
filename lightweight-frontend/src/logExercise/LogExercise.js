import React from "react";
import Dropdown from "react-bootstrap/Dropdown";
import { Link } from "react-router-dom";
import LightWeightAPI from "../api/api";
import "./LogExercise.css";

const Logexercise = ({
	loggedexerciseId,
	name,
	weight,
	unit,
	no_of_sets,
	no_of_reps,
	i,
	onLoggedExerciseDelete,
}) => {
	async function deleteLoggedExercise(loggedexerciseId) {
		await LightWeightAPI.deleteLoggedExercise(loggedexerciseId);
		onLoggedExerciseDelete(loggedexerciseId);
	}

	return (
		<div className="d-flex flex-row border m-3">
			<div className="mr-1">#{i + 1}</div>
			<div>
				<Dropdown>
					<Dropdown.Toggle variant="blue" id="dropdown-basic"></Dropdown.Toggle>
					<Dropdown.Menu>
						<Dropdown.Item>
							<Link
								to={`/workouts/edit/${loggedexerciseId}/${weight}/${unit}/${no_of_sets}/${no_of_reps}`}
								className="LogExercise-edit-button"
							>
								Edit
							</Link>
						</Dropdown.Item>
						<Dropdown.Item
							onClick={() => deleteLoggedExercise(loggedexerciseId)}
						>
							Delete
						</Dropdown.Item>
					</Dropdown.Menu>
				</Dropdown>
			</div>

			<div className="flex-grow-1 ml-5">
				<p>
					<strong>{name}</strong>
				</p>
				<p>
					{no_of_sets} sets x {no_of_reps} reps ({weight}
					{unit})
				</p>
			</div>
			<div>
				<Link to={`exercise/${name}`}>
					<button
						className="btn btn-sm"
						style={{ backgroundColor: "#461111", color: "white" }}
					>
						View
					</button>
				</Link>
			</div>
		</div>
	);
};

export default Logexercise;
