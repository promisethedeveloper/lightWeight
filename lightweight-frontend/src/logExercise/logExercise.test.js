import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LogExercise from "./LogExercise";

/** Smoke test */
it("Should render without crashing", () => {
	render(
		<MemoryRouter>
			<LogExercise />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<LogExercise />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders logExercise page with 'View' button", () => {
	const { getByText } = render(
		<MemoryRouter>
			<LogExercise />
		</MemoryRouter>
	);
	expect(getByText("View")).toBeInTheDocument();
});
