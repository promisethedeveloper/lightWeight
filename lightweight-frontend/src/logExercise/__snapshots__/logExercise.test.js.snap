// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-row border m-3"
  >
    <div
      class="mr-1"
    >
      #NaN
    </div>
    <div>
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          class="dropdown-toggle btn btn-blue"
          id="dropdown-basic"
          type="button"
        />
      </div>
    </div>
    <div
      class="flex-grow-1 ml-5"
    >
      <p>
        <strong />
      </p>
      <p>
         sets x  reps ()
      </p>
    </div>
    <div>
      <a
        href="/exercise/undefined"
      >
        <button
          class="btn btn-sm"
          style="background-color: rgb(70, 17, 17); color: white;"
        >
          View
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
